python/lsst/db/db.py

```diff
@@ -54,547 +41,115 @@
 
-        read_default_file
-          file from which default client values are read
+        @param driverName Name of the driver, in a form accepted by SQLAlchemy. See:
+               http://docs.sqlalchemy.org/en/latest/core/engines.html#database-urls
+               for more details. Examples: "mysql", "mysql+mysqldb, "sqlite",
+               "postgresql"
 
-        charset
-          If supplied, the connection character set will be changed
-          to this character set (MySQL-4.1 and newer). This implies
-          use_unicode=True.
-
-        local_infile
-          integer, non-zero enables LOAD LOCAL INFILE; zero disables
-
-        Note that if the host is set to "localhost", "127.0.0.1" will be used
-        instead, because MySQL silently switches to a default socket if
-        "localhost" is used.
+        @param logFlag    Flag turning on/off SQLAlchemy logging.
         """
-        self._conn = None
-        self._log = log.getLogger("lsst.db.Db")
-        self._kwargs = copy.deepcopy(kwargs)
-        self._sleepLen = self._kwargs.pop("sleepLen", 3)
-        self._attemptMaxNo = 1 + self._kwargs.pop("maxRetryCount", 0)
-        for k in self._kwargs:
-            if k not in self.connectArgToOptionMap:
-                raise DbException(DbException.INVALID_CONN_INFO, repr(k))
-        # If a MySQL defaults file will be read, make sure to pull values from the
-        # [mysql] group, just like the mysql executable does. The [client] group
-        # will automatically be read.
-        if "read_default_file" in self._kwargs:
-            f = os.path.expanduser(self._kwargs["read_default_file"])
-            self._kwargs["read_default_file"] = f
-            if not os.path.isfile(f):
-                raise DbException(DbException.INVALID_OPT_FILE, f)
-            self._kwargs["read_default_group"] = "mysql"
-        # MySQL will use socket for "localhost". 127.0.0.1 forces TCP.
-        if self._kwargs.get("host", "") == "localhost":
-            self._kwargs["host"] = "127.0.0.1"
-            self._log.warning('"localhost" specified, switching to 127.0.0.1')
-        if "port" in self._kwargs:
-            self._kwargs["port"] = int(self._kwargs["port"])
-        # Map MySQL warnings to exceptions
-        warnings.filterwarnings("error", category=MySQLdb.Warning)
-        # Log the connection parameters
-        self._log.info("Created lsst.db.Db with connection parameters " + \
-                       "(password not shown): %s", \
-                       str(["%s:%s" % (x, self._kwargs[x]) \
-                            for x in self._kwargs if not x == "passwd"]))
+        self._driverName = driverName
+        self._logFlag = logFlag
 
-    def __del__(self):
+    def getEngineFromFile(self, fileName):
         """
-        Disconnect from the server.
-        """
-        self.disconnect()
+        Initializes and returns SQLAlchemy engine using values provided through the
+        file.
```

Comment:
Couple of words about config file format? Also mention explicitly here that this method only supports mysql drivers.